src/admin/admin.routes.tsx

```diff
@@ -1,35 +1,19 @@
-import React, { ReactNode } from 'react';
+import React from 'react';
+import { Switch } from 'react-router';
 
-import { SecuredRoute } from '../authentication/components';
+import { renderAuthenticationRoutes } from '../authentication/authentication.routes';
+import { renderErrorRoutes } from '../error/error.routes';
+import { renderAdminContentRoutes } from './content/content.routes';
+import { renderAdminDashboardRoutes } from './dashboard/dashboard.routes';
+import { renderAdminMenuRoutes } from './menu/menu.routes';
 
-import { ADMIN_PATH } from './admin.const';
-import { Dashboard, MenuDetail, MenuEdit, MenuOverview } from './views';
-
-export const renderAdminRoutes = (): ReactNode[] => [
-	<SecuredRoute
-		key={ADMIN_PATH.DASHBOARD}
-		component={Dashboard}
-		exact
-		path={ADMIN_PATH.DASHBOARD}
-	/>,
-	// Menu routes
-	<SecuredRoute key={ADMIN_PATH.MENU} component={MenuOverview} exact path={ADMIN_PATH.MENU} />,
-	<SecuredRoute
-		key={ADMIN_PATH.MENU_DETAIL}
-		component={MenuDetail}
-		exact
-		path={ADMIN_PATH.MENU_DETAIL}
-	/>,
-	<SecuredRoute
-		key={ADMIN_PATH.MENU_CREATE}
-		component={MenuEdit}
-		exact
-		path={ADMIN_PATH.MENU_CREATE}
-	/>,
-	<SecuredRoute
-		key={ADMIN_PATH.MENU_EDIT}
-		component={MenuEdit}
-		exact
-		path={ADMIN_PATH.MENU_EDIT}
-	/>,
-];
+export const renderAdminRoutes = () => (
+	<Switch>
+		{renderAdminDashboardRoutes()}
+		{renderAdminMenuRoutes()}
+		{renderAdminContentRoutes()}
+		{/* Default routes */}
+		{renderAuthenticationRoutes()}
```

Comment:
voeg nog een todo toe om hier ook een permissie te checken: has_access_to_admin_dashboard

Reply:
Die `renderAuthenticationRoutes` moet daar zelfs niet tussen staan.
Er staat al in de App.tsx een todo om te voorkomen dat ze naar de admin kunnen gaan als ze niet de juiste permissies hebben.